docs/manual/philosophy.md

```diff
@@ -0,0 +1,148 @@
+
+There are a few command line options to allow users influence how
+packages are installed but their use is discouraged for the most part
+and packages should be installed as they are.
+
+
+## RPM in the Software handling stack
+
+RPM sees itself as a specific layer in the software handling stack. It
+relies on build systems like make and the upstream provided build
+scripts to orchestrate the actual build and passes the acquiring and
+selection of the "right" packages up to updaters like yum, dnf,
+zypper, etc.
+
+The stack typically looks like this:
+
+ * Upstream build scripts using tools like maker, cmake, ant, ...
+ * rpmbuild for running those via a Spec file
+ * Build systems for installing build dependencies and keeping track of build artifacts
+ * Package repositories offering binary packages
+ * Installers/Updaters selecting and downloading packages
+ * rpm checking for consistency and installing/updating the packages
+
+## Design goals
+
+### Rolling out (security) updates quickly.
+
+Getting updates installed quickly is one of the main design
+goals. Many features work towards this:
+
+#### Packaging dependencies separately
+
+Libraries should be packaged separately and binaries should link to
+the version provided by system packages. Static linking is
+discouraged. This limits the number of packages that need updates or
+re-builds in case of a vulnerability.
+
+#### Unattended operation
+
+Package installation and update is unattended and must not require
+user interaction. If possible delay user interaction to the first
+start up.
+
+#### Clear update path
+
+Each package name creates an update path where packages with the same
+name and a newer version (technically Epoch-Version-Release) are an
+update for the packages with lower version. Packages are not allowed
+to make assumptions on what intermediate packages got installed or
+not.
+
+#### Build Sources, Changes and Build instructions together
+
+Source packages bundle upstream sources, patches and build
+instructions into one entity. If changes need to be made everything
+needed is available, including a list of packages needed to run the
+build.
+
+Nosource packages are a exception that does not ship the sources but
+only the build instructions to keep proprietary source private. They
+still can be used to generate binary packages in combination with the
+sources.
+
+#### Separate Upstream Source from Patches
+
+Source packages are supposed to contain the unaltered upstream sources
+in a way their checksum can be checked. All changes can be done by
+applying patches or running build scripts. This makes it easy to
+understand what the packager did to the software. This is important to
+be able to figure out where issue arise from and to see which issues
+are fixed in the package - even if the upstream version may still be
+vulnerable.
+
+#### Support back porting fixes
+
+Distribution often have a different approach to fixing software than
```

Comment:
Copy edit: *A* distribution often *has* a different approach to fixing software *from an* upstream project.